src/shared/components/query/GenesetsVolcanoSelector.tsx

```diff
@@ -0,0 +1,313 @@
+    readonly percentileOptions = [{label: '50%', value: '50'}, {label: '75%', value: '75'}, {label: '100%', value: '100'}];
+    @observable private map_genesets_selected = new ObservableMap<boolean>();
+    constructor(props:GenesetsVolcanoSelectorProps)
+    {
+        super(props);
+        this.state = {plotData: []};
+        this.percentileChange = this.percentileChange.bind(this);
+        this.updateSelectionFromPlot = this.updateSelectionFromPlot.bind(this);
+    }
+    
+    componentDidMount()
+    {
+        this.fetchData();
+    }
+    
+    componentDidUpdate()
+    {
+        if (this.percentileHasChanged) {
+            this.isLoading = true;
+            this.fetchData();
+            this.percentileHasChanged = false;
+        }
+    }
+    
+    private columns:IColumnDefMap = {
+            'genesets': {
+                priority: 1,
+                dataField: 'name',
+                name: "Gene Sets",
+                sortable: true,
+                filterable: true,
+            },
+            'gsvaScore': {
+                priority: 2,
+                dataField: 'representativeScore',
+                name: "GSVA Score",
+                sortable: true,
+                filterable: true,
+                formatter: ({name, columnData: value}: IColumnFormatterData<Geneset>) => (
+                        <Td key={name} column={name} value={value}>
+                        {value.toFixed(2)}
+                        </Td>
+                    ),
+            },
+            'pValue': {
+                priority: 3,
+                dataField: 'representativePvalue',
+                name: "P Value",
+                sortable: true,
+                filterable: true,
+                formatter: ({name, columnData: value}: IColumnFormatterData<Geneset>) => (
+                    <Td key={name} column={name} value={value}>
+                    {toPrecision(value, 2, 0.1)}
+                    </Td>
+                ),
+            },
+            'selected': {
+                name: "Selected",
+                priority: 4,
+                sortable: false,
+                filterable: false,
+                formatter: ({name, rowData: geneset}:IColumnFormatterData<Geneset>) => (
+                    <Td key={name} column={name}>
+                    {!!(geneset) && (
+                            <div className={styles.selectionColumnCell}>
+                            <Observer>
+                            {() => (
+                                <LabeledCheckbox
+                                checked={!!this.map_genesets_selected.get(geneset.name)}
+                                onChange={event => (this.map_genesets_selected.set(geneset.name, event.target.checked)) }
+                            />
+                            )}
+                    </Observer>
+                    </div>
+                    )}
+                    </Td>
+                ),
+            },
+    };
+    
+    async fetchData() {
```

Comment:
i think this should live in queryStore.  State/data should not be managed in component.

Reply:
And we should use a mobxPromise to handle loading state.